### PR TITLE
Add guard on JBoss special handling and process fact prefixes. Closes…

### PR DESCRIPTION
**Pull request by chambridge**
… #120.

### CODE REVIEW

File: library/spit_results.py [hunk 1]
```diff
@@ -251,17 +251,28 @@ def write_to_csv(self):
...
-        for _, host_vars in iteritems(self.all_vars):
-            uuid = host_vars['connection.uuid']
-            host_vals = safe_next((vals
-                                   for vals in self.vals
-                                   if vals['connection.uuid'] == uuid))
-
-            host_vals.update(process_jboss_versions(host_vars))
-            host_vals.update(process_addon_versions(host_vars))
+        fact_keys = ['connection.host', 'connection.port', 'connection.uuid']
+        if isinstance(self.fact_names, list):
```
Comment (noahl) [inline library/spit_results.py:255]: When would `fact_names` not be a list?

File: library/spit_results.py [hunk 2]
```diff
@@ -251,17 +251,28 @@ def write_to_csv(self):
...
-        # Special processing for JBoss facts.
-        for _, host_vars in iteritems(self.all_vars):
-            uuid = host_vars['connection.uuid']
-            host_vals = safe_next((vals
-                                   for vals in self.vals
-                                   if vals['connection.uuid'] == uuid))
-
-            host_vals.update(process_jboss_versions(host_vars))
-            host_vals.update(process_addon_versions(host_vars))
+        fact_keys = ['connection.host', 'connection.port', 'connection.uuid']
```
Comment (noahl) [inline library/spit_results.py:254]: I'd rather move this logic to `rho/utilities.py` with the other fact names, but I can do that in a separate PR if you like.

**Review by noahl — APPROVED** [review]
Two comments, but LGTM.

File: library/spit_results.py (hunk 2 above)
Comment (chambridge) [reply library/spit_results.py:254]: Will move this logic to rho/utilities when I also remove the annoying fact group prefixing.

File: library/spit_results.py (hunk 1 above)
Comment (chambridge) [reply library/spit_results.py:255]: should always be, moved this code over from run_cmds.py so I could unblock my merge conflicts. This section of code will go away when I remove the fact group prefixing (i.e. Username_uname.os; will get rid of Username_). Then self.facts_names should be what we want.

**Pull request merged**